Tests/Packet++Test/Tests/IcmpV6Tests.cpp

```diff
@@ -0,0 +1,287 @@
+	pcpp::ICMPv6EchoLayer *echoRequestLayer = echoRequestPacket.getLayerOfType<pcpp::ICMPv6EchoLayer>();
+	PTF_ASSERT_NOT_NULL(echoRequestLayer);
+	PTF_ASSERT_TRUE(echoRequestLayer->isMessageOfType(pcpp::ICMPv6MessageType::ICMPv6_ECHO_REQUEST));
+	PTF_ASSERT_EQUAL((int)echoRequestLayer->getMessageType(), 128);
+	PTF_ASSERT_EQUAL(echoRequestLayer->getCode(), 0);
+	PTF_ASSERT_EQUAL(echoRequestLayer->getChecksum(), 0x7a4c);
+	PTF_ASSERT_EQUAL(echoRequestLayer->getIdentifier(), 0x0018);
+	PTF_ASSERT_EQUAL(echoRequestLayer->getSequenceNr(), 20);
+	PTF_ASSERT_EQUAL(echoRequestLayer->getEchoDataLen(), 56);
+
+	uint8_t data[] = {0xbd, 0xce, 0xcb, 0x62};
+	PTF_ASSERT_BUF_COMPARE(echoRequestLayer->getEchoDataPtr(), data, 4);
+
+	// Echo reply
+	PTF_ASSERT_TRUE(echoReplyPacket.isPacketOfType(pcpp::ICMPv6));
+	pcpp::ICMPv6EchoLayer *echoReplyLayer = echoReplyPacket.getLayerOfType<pcpp::ICMPv6EchoLayer>();
+	PTF_ASSERT_NOT_NULL(echoReplyLayer);
+	PTF_ASSERT_TRUE(echoReplyLayer->isMessageOfType(pcpp::ICMPv6MessageType::ICMPv6_ECHO_REPLY));
+	PTF_ASSERT_EQUAL((int)echoReplyLayer->getMessageType(), 129);
+	PTF_ASSERT_EQUAL(echoReplyLayer->getCode(), 0);
+	PTF_ASSERT_EQUAL(echoReplyLayer->getChecksum(), 0x794c);
+	PTF_ASSERT_EQUAL(echoReplyLayer->getIdentifier(), 0x0018);
+	PTF_ASSERT_EQUAL(echoReplyLayer->getSequenceNr(), 20);
+	PTF_ASSERT_EQUAL(echoReplyLayer->getEchoDataLen(), 56);
+
+	PTF_ASSERT_BUF_COMPARE(echoReplyLayer->getEchoDataPtr(), data, 4);
+
+	// Neighbor solicitation with source link-layer option
+	PTF_ASSERT_TRUE(neighSoliPacket.isPacketOfType(pcpp::ICMPv6));
+	pcpp::NDPNeighborSolicitationLayer *neighSoliLayer = neighSoliPacket.getLayerOfType<pcpp::NDPNeighborSolicitationLayer>();
+	PTF_ASSERT_NOT_NULL(neighSoliLayer);
+	PTF_ASSERT_EQUAL(neighSoliLayer->getHeaderLen(), 32);
+	PTF_ASSERT_EQUAL((int)neighSoliLayer->getMessageType(), 135);
+	PTF_ASSERT_EQUAL(neighSoliLayer->getCode(), 0);
+	PTF_ASSERT_EQUAL(neighSoliLayer->getChecksum(), 0xfe98);
+	PTF_ASSERT_EQUAL(neighSoliLayer->getTargetIP(), pcpp::IPv6Address("fd53:7cb8:383:2::1:117"));
+	PTF_ASSERT_TRUE(neighSoliLayer->hasLinkLayerAddress());
+	PTF_ASSERT_EQUAL(neighSoliLayer->getLinkLayerAddress(), pcpp::MacAddress("00:54:af:e9:4d:80"));
+	PTF_ASSERT_EQUAL(neighSoliLayer->getNdpOptionCount(), 1);
+
+	pcpp::NdpOption sourceLinkLayerOption = neighSoliLayer->getNdpOption(pcpp::NDPNeighborOptionTypes::NDP_OPTION_SOURCE_LINK_LAYER);
+	PTF_ASSERT_TRUE(sourceLinkLayerOption.isNotNull());
+	PTF_ASSERT_EQUAL(sourceLinkLayerOption.getNdpOptionType(), pcpp::NDPNeighborOptionTypes::NDP_OPTION_SOURCE_LINK_LAYER);
+	PTF_ASSERT_EQUAL(sourceLinkLayerOption.getDataSize(), 6);
+	PTF_ASSERT_EQUAL(sourceLinkLayerOption.getTotalSize(), 8);
+	PTF_ASSERT_EQUAL(pcpp::MacAddress(sourceLinkLayerOption.getValue()), pcpp::MacAddress("00:54:af:e9:4d:80"));
+	pcpp::NdpOption sourceLinkLayerOption2 = neighSoliLayer->getNdpOption(pcpp::NDPNeighborOptionTypes::NDP_OPTION_TARGET_LINK_LAYER);
```

Comment:
nit: we can call this variable `targetLinkLayerOption`, no?

Reply:
@kolbex you resolved a few comments but didn't push any commits, maybe you forgot to `git push`? 🤔

Reply:
sorry for the late push